core/Pos/Core/Genesis/Types.hs

```diff
@@ -37,17 +46,35 @@ data StakeDistribution
         , sdPoorStake :: !Coin
         }
     -- | First three nodes get 0.875% of balance.
-    | ExponentialStakes !Word
+    | ExponentialStakes !Word -- ^ Numbers of participants
+                        !Coin -- ^ Minimal coin
     -- | Custom balances list.
     | CustomStakes [Coin]
     deriving (Show, Eq, Generic)
 
+-- | Map each participant coin in the distribution.
+stakeDistrMapCoin :: (Coin -> Coin) -> StakeDistribution -> StakeDistribution
+stakeDistrMapCoin f d = case d of
+    (FlatStakes n c)       -> FlatStakes n (applyN n c)
```

Comment:
I don't like it. It works only if `f(c / n) = f^n(c) / n` for all `c, n`. Which is true if `f` adds a constant, but is wrong if `f` is e. g. `unsafeMulCoin 3`.